app/models/llms/basic_workflow/integration.py

```diff
@@ -0,0 +1,391 @@
+                
+                first_choice = response.choices[0]
+                if not hasattr(first_choice, 'message') or not first_choice.message:
+                    raise RuntimeError("No message in API response choice")
+                
+                if not hasattr(first_choice.message, 'content') or not first_choice.message.content:
+                    raise RuntimeError("No content in API response message")
+                
+                itinerary = first_choice.message.content.strip()
+                return itinerary
+                
+            except Exception as e:
+                error_message = str(e)
+                
+                if "authentication" in error_message.lower() or "api_key" in error_message.lower():
+                    raise ValueError("Authentication failed. Please check your OpenAI API key.")
+                elif "rate limit" in error_message.lower():
+                    raise RuntimeError("Rate limit exceeded. Please try again later.")
+                else:
+                    raise RuntimeError(f"OpenAI API error: {error_message}")
+
+    def get_days(self, user_input_lower: str) -> str:
+        import re
+        from num2words import num2words
+        
+        day_pattern = self.days_config["patterns"]["day_pattern"]
+        week_pattern = self.days_config["patterns"]["week_pattern"]
+        written_numbers = self.days_config["written_numbers"]
+        written_weeks = self.days_config["written_weeks"]
+        default_duration = self.days_config["default_duration"]
+        
+        day_match = re.search(day_pattern, user_input_lower)
+        if day_match:
+            days = int(day_match.group(1))
+            return f"{days} days"
+        
+        for i in range(1, 101):  # Check numbers 1-100
+            written_number = num2words(i).lower()
+            if f"{written_number} days" in user_input_lower or f"{written_number} day" in user_input_lower:
+                return f"{i} days"
+        
+        for word, number in written_numbers.items():
+            if f"{word} days" in user_input_lower or f"{word} day" in user_input_lower:
+                return f"{number} days"
+        
+        week_match = re.search(week_pattern, user_input_lower)
+        if week_match:
+            weeks = int(week_match.group(1))
+            days = weeks * 7
+            return f"{days} days"
+        
+        for i in range(1, 31):  # Check weeks 1-30
+            written_week = num2words(i).lower()
+            if f"{written_week} week" in user_input_lower or f"{written_week} weeks" in user_input_lower:
+                days = i * 7
+                return f"{days} days"
+            
+        for week_phrase, days in written_weeks.items():
+            if week_phrase in user_input_lower:
+                return f"{days} days"
+        
+        return default_duration
+
+    def get_destinations(self, user_input_lower: str) -> str:
```

Comment:
The method name `get_destinations` suggests fetching, but it parses/extracts destinations from text. Rename to `extract_destination_from_text`.